cmake/target/build.cmake

```diff
@@ -108,7 +75,11 @@ function(setup_build_module MODULE SOURCES GENERATED EXCLUDED_SOURCES DEPENDENCI
             add_dependencies(${MODULE} "${DEPENDENCY}")
             list(APPEND TARGET_DEPENDENCIES "${DEPENDENCY}")
         endif()
-        target_link_libraries(${MODULE} PUBLIC "${DEPENDENCY}")
+        # Avoid linking custom targets, and executables
+        is_target_library(IS_LIB "${DEPENDENCY}")
+        if (NOT TARGET "${DEPENDENCY}" OR IS_LIB)
```

Comment:
What are dependencies that aren't targets? I'm trying to understand what this case is trying to cover.

Reply:
You can have a dependency on executables to make them build as part of a deployment.  However, these should not be linked.  Utility targets (custom targets) behave similarly.

Reply:
If that's the case, couldn't this check just be `if (IS_LIB)`? What other than libraries are you trying to link against?

Reply:
@Joshua-Anderson IS_LIB returns FALSE in the case that the TARGET does not exist. However, you can specify link dependencies before actually creating those libraries.  Thus, this code does the following:

1. If the dependency is not yet a TARGET assume it is a library (common case) and link against it.
2. If the dependency is a a library (defined as a library type) also link against it.

Reply:
Sorry, I misunderstood the reason behind this code.  Here is a clearer explanation.  We need to call `target_link_libraries` on two types of items:

1. Linker flags (e.g. -lrt)
2. Defined libraries

It is the link flags that are not targets and thus pass the NOT TARGET check.  Perhaps this should be implemented as a "starts with `-l`" check

Reply:
I cleaned this up and documented what is going on with a few local comments, better abstraction, and an explanation of caveats with `register_fprime_executable`.

Note: we have one other thing to link

3. Undefined tagets (that will be defined) are assumed to be libraries for backwards compatibility. Thus, `add_dependencies(TARGET EXECUTABLE)` is preferred to MOD_DEPS to build executables as dependencies.

The above is now documented.